source/aggregation/agg-exp-ops.txt

```diff
@@ -0,0 +1,1183 @@
+
+The examples in this guide assume that you include the following imports
+in your code:
+
+.. code-block:: kotlin
+   :copyable: true
+
+   import com.mongodb.client.model.Aggregates
+   import com.mongodb.client.model.Accumulators
+   import com.mongodb.client.model.Projections
+   import com.mongodb.client.model.Filters
+   import com.mongodb.client.model.mql.MqlValues
+
+To access document fields in an expression, you must reference the
+current document being processed by the aggregation pipeline by using the
+``current()`` method. To access the value of a
+field, you must use the appropriately typed method, such as
+``getString()`` or ``getDate()``. When you specify the type for a field,
+you ensure that the driver provides only those methods which are
+compatible with that type. The following code shows how to reference a
+string field called ``name``:
+
+.. code-block:: kotlin
+   :copyable: true
+
+   current().getString("name")
+
+To specify a value in an operation, pass it to the ``of()`` constructor method to
+convert it to a valid type. The following code shows how to reference a
+value of ``1.0``:
+
+.. code-block:: kotlin
+   :copyable: true
+
+   of(1.0)
+
+To create an operation, chain a method to your field or value reference.
+You can build more complex operations by chaining other methods.
+
+The following example creates an operation to find patients in New
+Mexico who have visited the doctor’s office at least once. The operation
+performs the following actions:
+
+- Checks if the size of the ``visitDates`` array is greater than ``0``
+  by using the ``gt()`` method
+- Checks if the ``state`` field value is “New Mexico” by using the
+  ``eq()`` method
+
+The ``and()`` method links these operations so that the pipeline stage
+matches only documents that meet both criteria.
+
+.. code-block:: kotlin
+   :copyable: true
+
+   current()
+       .getArray("visitDates")
+       .size()
+       .gt(of(0))
+       .and(current()
+           .getString("state")
+           .eq(of("New Mexico")))
+
+While some aggregation stages, such as ``group()``, accept operations
+directly, other stages expect that you first include your operation in a
+method such as ``computed()`` or ``expr()``. These methods, which take
+values of type ``TExpression``, allow you to use your expressions in
+certain aggregations.
+
+To complete your aggregation pipeline stage, include your expression
+in an aggregates builder method. The following list provides examples of
+how to include your expression in common aggregates builder methods:
+
+- ``match(expr(<expression>))``
+- ``project(fields(computed("<field name>", <expression>)))``
+- ``group(<expression>)``
+
+.. TODO To learn more about these methods, see the
+.. :ref:`kotlin-sync-sync-aggregation`.
+
+The examples use the ``listOf()`` method to create a list of
+aggregation stages. Then, the examples pass the pipeline to the
+``aggregate()`` method of ``MongoCollection``.
+
+Constructor Methods
```

Comment:
When we first developed this page for Java sync, we discussed this a lot. I think we should leave as is, because this table describes the functionality of a kotlin/java-native API